pubsub/driver/driver.go

```diff
@@ -78,6 +78,11 @@ type Message struct {
 	AfterSend func(asFunc func(interface{}) bool) error
 }
 
+// Implement batcher item for byte sized based batching
+func (m *Message) ByteSize() int {
+	return len(m.Body)
```

Comment:
Is it just the `Body` that's included? What about the attributes?

Reply:
Yes only the `Data` field is contributing, the `Attributes` one (from Metadata field) is subject to other limitations as described in https://cloud.google.com/pubsub/quotas#resource_limits which can probably be left as "server side" validation errors.

Reply:
However I can add an estimated size based on the attributes as well so it contributes to the batching, even though GCP doesn't really care, since that limitation could technically be used for other providers.